cmd/server-mux.go

```diff
@@ -194,6 +199,9 @@ func newListenerMux(listener net.Listener, config *tls.Config) *ListenerMux {
 				return
 			}
 
+			// Enable Read timeout
+			conn.SetReadDeadline(time.Now().Add(defaultTcpReadTimeout))
+
 			// Enable keep alive for each connection.
```

Comment:
> What we need is doing "conn" list ourselves i.e basically make sure that we don't run out of RLimit.Max and handle it ourselves..

This is not only about the maximum opened files but also about having limited port range (< 65536) 

> So that we can reject new connection attempts gracefully.

Probably this is more than luxury, this is a rare scenario and our users will already try to avoid being limited by port numbers by using other methods.